src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+                onChange(value as string);
+                setNamedPolicyDropdownOpen(false);
+              }}
+              selections={value}
+              variant={SelectVariant.single}
+            >
+              <SelectOption
+                key={0}
+                data-testid="persistent-option"
+                value={t("persistent")}
+                isPlaceholder
+              />
+              <SelectOption
+                key={1}
+                data-testid="transient-option"
+                value={t("transient")}
+              />
+              <SelectOption
+                key={2}
+                data-testid="email-option"
+                value={t("email")}
+              />
+              <SelectOption
+                key={3}
+                data-testid="kerberos-option"
+                value={t("kerberos")}
+              />
+              <SelectOption
+                key={4}
+                data-testid="x509-option"
+                value={t("x509")}
+              />
+              <SelectOption
+                key={5}
+                data-testid="windowsDomainQN-option"
+                value={t("windowsDomainQN")}
+              />
+              <SelectOption
+                key={6}
+                data-testid="unspecified-option"
+                value={t("unspecified")}
+              />
+            </Select>
+          )}
+        ></Controller>
+      </FormGroup>
+
+      <FormGroup
+        label={t("principalType")}
+        labelIcon={
+          <HelpItem
+            helpText={th("principalType")}
+            forLabel={t("principalType")}
+            forID="kc-principalType"
+          />
+        }
+        fieldId="kc-principalType"
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.principalType"
+          defaultValue={t("subjectNameId")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-principalType"
+              onToggle={() =>
+                setPrincipalTypeDropdownOpen(!principalTypeDropdownOpen)
+              }
+              isOpen={principalTypeDropdownOpen}
+              onSelect={(_, value) => {
+                onChange(value as string);
+                setPrincipalTypeDropdownOpen(false);
+              }}
+              selections={value}
+              variant={SelectVariant.single}
+            >
+              <SelectOption
+                key={0}
+                data-testid="subjectNameId-option"
+                value={t("subjectNameId")}
+                isPlaceholder
+              />
+              <SelectOption
+                key={1}
+                data-testid="attributeName-option"
+                value={t("attributeName")}
+              />
+              <SelectOption
+                key={2}
+                data-testid="attributeFriendlyName-option"
+                value={t("attributeFriendlyName")}
+              />
```

Comment:
I don't think it's needed to give these `SelectOption` a `key` prop, as they are statically populated.
```suggestion
              <SelectOption
                data-testid="subjectNameId-option"
                value={t("subjectNameId")}
                isPlaceholder
              />
              <SelectOption
                data-testid="attributeName-option"
                value={t("attributeName")}
              />
              <SelectOption
                data-testid="attributeFriendlyName-option"
                value={t("attributeFriendlyName")}
              />
```

Reply:
yep, all keys were removed last iteration, ty